### PR TITLE
Add Doxygen list of libraries for the main overview

### PR DESCRIPTION
**Pull request by sarenameas**
- Add the list of libraries to the main page.
- Please see these PRs for the anchors:
  https://github.com/FreeRTOS/coreMQTT/pull/47
  https://github.com/FreeRTOS/coreJSON/pull/18
  https://github.com/aws/device-shadow-for-aws-iot-embedded-sdk/pull/13

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: docs/doxygen/pages.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
 /**
 @mainpage Overview
 @brief The AWS IoT Device SDK for Embedded C provides demonstration code and integration tests for a collection of libraries.
+
+The following libraries are part this SDK:
```
Comment (muneebahmed10) [inline docs/doxygen/pages.txt:5]: Should this be in a `@section` or `@subpage` so that it shows up in the sidebar?

**Review by muneebahmed10 — DISMISSED** [review]

File: docs/doxygen/pages.txt (hunk 1 above)
Comment (yourslab) [inline docs/doxygen/pages.txt:5]:
```suggestion
The following libraries are part of this SDK:
```

Comment (sarenameas) [reply docs/doxygen/pages.txt:5]: I didn;t think it needed to be there. But I guess it will look nice

Comment (sarenameas) [reply docs/doxygen/pages.txt:5]: thanks

**Review by yourslab — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**